### PR TITLE
Added Grammars

### PR DESCRIPTION
**Pull request by hilliardj**
Added the Grammar from the zip file submitted, and added a proposed grammar that removes left recursion.

### CODE REVIEW

**Comment by hilliardj** [comment]
@justiceadamsUNI @harshavarmak Let me know what you think of the updated grammar removing left recursion.  Thanks!

File: Compiler/doc/languageGrammar.txt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                        | <IDENTIFIER> ( <ACTUALS> )
+                        | <IDENTIFIER>
+                        | <LITERAL>
+                        | - <FACTOR>
+                        | ( <EXPR> )
+
+            <ACTUALS> ::= e
+                        | <NONEMPTYACTUALS>
+
+    <NONEMPTYACTUALS> ::= <EXPR>
```
Comment (justiceadamsUNI) [inline Compiler/doc/languageGrammar.txt:52]: This NONEMPTYACTUALS has  <EXP> on both sides of the | so it's ambiguous still. There's quite a few places this happens yet (SIMPLE-EXPR, TERM, NONEMPTYFORMALS)  So we need to fix that.

**Comment by justiceadamsUNI** [comment]
Any reason there's 3 different revised Grammars. I vote we just have orignalGrammar.txt and RevisedGrammar.txt. Then they can be edited anywhere with any editor. We can work on this tomorrow

**Comment by justice-adams** [comment]
We're gonna wait for the OK from wallingford after the status check to merge this

**Comment by justiceadamsUNI** [comment]
Waiting for wallingford to give the OK before we merge this

**Comment by justiceadamsUNI** [comment]
I read through this and I'm on board. Nice job @hilliardj . We can merge this once wallingford says it's good.

**Comment by harshavarmak** [comment]
@hilliardj I've changed two of the First set, the First(Program) and First(Definitions), it should be only {$}

**Pull request merged**